graphql/src/main/java/io/jmix/graphql/schema/scalar/coercing/FileRefCoercing.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.graphql.schema.scalar.coercing;
+
+import graphql.schema.CoercingParseValueException;
+import graphql.schema.CoercingSerializeException;
+import io.jmix.core.FileRef;
+import io.jmix.core.FileStorage;
+import io.jmix.core.metamodel.datatype.impl.FileRefDatatype;
+import io.jmix.graphql.service.FileService;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.FileNotFoundException;
+import java.util.AbstractMap;
+
+public class FileRefCoercing extends BaseScalarCoercing {
```

Comment:
Try to put the required beans as constructor arguments. And don't use static variable: `FileService.get()`. 
You can use the same approach as for `ScalarTypes#localTimeScalar`